### DISCUSSION TITLE
Retrieving the raw string of a Renderable

### DISCUSSION THREAD

**Discussion by jdc-pub** [Textualize/rich#1661] [Q&A] [answered]
Is there any way to retrieve the raw string of a `Renderable`?

For example, say I have some padding `p = Padding("[green[A message", (0,2))` which will print two spaces before the text. I could accomplish this simple example with something like `rich.print("  [green]A message")`.

Is there any supported way to retrieve what _would_ go in the `rich.print` statement to equivalently print other `Renderable`s?

**Comment by paw-lu** [comment, marked as answer]
Probably the easiest way to do this is to capture the output via `io.StringIO`


```python
import io

from rich import console

con = console.Console(file=io.StringIO(), force_terminal=True)
con.print("[bold red]Hello[/] World")
str_output = con.file.getvalue()

print(f"  {str_output}")
```

[Docs for more info.](https://rich.readthedocs.io/en/latest/console.html#capturing-output)

**Comment by willmcgugan** [comment]
See also https://rich.readthedocs.io/en/latest/console.html#capturing-output
